src/boot/efi/pe.c

```diff
@@ -162,11 +164,70 @@ static bool pe_section_name_equal(const char *a, const char *b) {
         return true;
 }
 
-static void pe_locate_sections(
+static bool pe_use_this_dtb(
+                const PeSectionHeader section_table[],
+                size_t n_section_table,
+                const void *dtb,
+                size_t dtb_size,
+                const Device *device_table,
+                size_t device_table_size,
+                size_t section_nb) {
+
+        assert(section_table);
+        assert(n_section_table > 0);
+        assert(dtb);
+
+        EFI_STATUS err;
+
+        static const Device *cached_device_table = NULL;
+        static const Device *cached_device = NULL;
```

Comment:
my hope was actually that we can get rid of these static fields, i.e. pass in the right device directly, instead of the whole table.

Reply:
That is, to do HWID calculations in the pe_locate_sections() after locating .hwids? Sure, will do

Reply:
yeah, in the pe_locate_sections() call check if you need it because te firmware doesn't have a dt compatible string, and then locate the right device, and just pass that in to the internal version, so that it just checks there. I think that would be the most elegant: no static variables, nice runtime behaviour and nice abstraction.